### PR TITLE
Upgrade to Dart 3

### PR DESCRIPTION
**Pull request by JaffaKetchup**

### CODE REVIEW

**Comment by provokateurin** [comment]
LGTM, but you need to adjust the Dart SDK version as the http 1.0.0 changelog mentions Dart 3 is now required.

**Comment by JaffaKetchup** [comment]
@provokateurin Whoops, forgot about that!

File: README.md
Comment (provokateurin) [inline README.md:1]: Unrelated changes, please remove

File: analysis_options.yaml
Comment (provokateurin) [inline analysis_options.yaml:1]: Unrelated changes, please remove

File: example/osm_nominatim_example.dart
Comment (provokateurin) [inline example/osm_nominatim_example.dart:1]: Unrelated changes, please remove

File: pubspec.yaml [hunk 1]
```diff
@@ -1,12 +1,13 @@
 name: osm_nominatim
 description: A library to perform OSM Nominatim searches also supporting reverse searches
-version: 2.0.1
+version: 3.0.0
```
Comment (provokateurin) [inline pubspec.yaml]: Please don't bump the version

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: analysis_options.yaml
Comment (JaffaKetchup) [reply analysis_options.yaml:1]: This is related. It's a requirement of Dart 3 and a close translation of the existing linting rules.

File: example/osm_nominatim_example.dart
Comment (JaffaKetchup) [reply example/osm_nominatim_example.dart:1]: This is related. It's a requirement of Dart 3 and the translation of the linting rules.

File: README.md
Comment (JaffaKetchup) [reply README.md:1]: Resolved in 79db714.

File: pubspec.yaml (hunk 1 above)
Comment (JaffaKetchup) [reply pubspec.yaml]: Resolved in 79db7145d2f545df38a39d9bd89b17b6fd8c42e0.

File: example/osm_nominatim_example.dart
Comment (provokateurin) [reply example/osm_nominatim_example.dart:1]: No, `Future` and `Future<void>` are the same thing (I think you switched up the comment)

Comment (JaffaKetchup) [reply example/osm_nominatim_example.dart:1]: Specifying `<void>` is a requirement in Dart 3 of the strict rules (specifically `strict-raw-casts`) that I added as a rough translation of the old ones. I've removed it in 9f4b2c6.

It is recommended by the Dart Style Guidelines: https://dart.dev/effective-dart/design#do-use-futurevoid-as-the-return-type-of-asynchronous-members-that-do-not-produce-values.

**Comment by JaffaKetchup** [comment]
Requested changes have been made, although I do disagree with the two lint/(related) changes (https://github.com/provokateurin/osm-nominatim/pull/8#discussion_r1209443885).

**Comment by JaffaKetchup** [comment]
Hey @provokateurin, hope this can be merged soon!

**Comment by provokateurin** [comment]
Sorry, I forgot about this 😅

**Comment by QTree-app** [comment]
:)

**Comment by provokateurin** [comment]
Sorry again :sweat:

**Comment by lucasdidur** [comment]
Any update about this? @provokateurin

**Review by provokateurin — APPROVED** [review]

**Comment by provokateurin** [comment]
Sorry for taking so long. The change is published in `3.0.0`.
